code/html/custom.js

```diff
@@ -164,21 +168,26 @@ function validatePassword(password) {
     );
 }
 
-function validateForm(form) {
+function validateFormPasswords(form) {
+    var passwords = $("input[name='adminPass'],input[name='adminPass_confirm']", form);
 
-    // password
-    var adminPass1 = $("input[name='adminPass']", form).first().val();
-    if (!validatePassword(adminPass1)) {
+    var adminPass1 = passwords.first().val(),
+        adminPass2 = passwords.last().val();
+
+    if ((adminPass1.length > 0) && !validatePassword(adminPass1)) {
```

Comment:
It would skip this check if `adminPass1` was an empty string right?

Reply:
True, I was following original check logic. On '#password' layout it falls through, sends the data and returns 'No changes detected' from device.

General '#layout' can have empty password fields, no need to alert. But I think '#password' should have 'required' attrs on inputs, so this check becomes 'required' && length>0 or just 'form/input .checkValidity()' (which btw I remember finding out about and briefly trying it at #888 - form validation without alerts, just highlighting hints)

Reply:
I totally think going for the HTML5 input validation being the best bet! This validation is still JS and can be turned off by the user so ideally there should be some backend validation going on too.